src/bindings/python/src/pyopenvino/graph/shape.cpp

```diff
@@ -44,9 +44,30 @@ void regclass_graph_Shape(py::module m) {
         if (key < 0) {
             key += v.size();
         }
-        return v[key];
+        if (key >= 0 && key < static_cast<int64_t>(v.size())) {
```

Comment:
Using v.at(key) will throw if key out of range.
ov::Shape accepts negative index the normalization above is not required.